File: DSA/PranyaGupta_500126032/Day34Question1.py
class Solution:
    def addOne(self, n):
        i=0
        while n:
            if (n&1<<i) != 0:
                n^= (1<<i)
                i+=1
            else:
                n^=(1<<i)
                return n    

def main():
    n = int(input("Enter the number: "))
    sol = Solution()
    result = sol.addOne(n)
    print(result)  
if __name__ == "__main__":
    main()

File: DSA/Gautam_Shyamnani_590013078/Day34Question1.cpp
#include <iostream>
using namespace std;

int addOne(int n) {
    int m = 1;

    
    while (n & m) {
        n = n ^ m;      
        m = m << 1;     
    }

    n = n ^ m;         

    return n;
}

int main() {
    int n = 7; 
    cout << "Result: " << addOne(n) << endl;
    return 0;
}
